thunderbird/processes/wps_generate_climos.py

```diff
@@ -0,0 +1,296 @@
+            ComplexOutput(
+                "dry_output",
+                "Dry Output",
+                as_reference=True,
+                abstract="File information",
+                supported_formats=[FORMATS.TEXT],
+            ),
+        ]
+
+        super(GenerateClimos, self).__init__(
+            self._handler,
+            identifier="generate_climos",
+            version="0.8.0",
+            title="Generate Climatological Means",
+            abstract="Generate files containing climatological means from input files of daily, monthly, or yearly data that adhere to the PCIC metadata standard (and consequently to CMIP5 and CF standards).",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def dry_run_info(self, filepath, climo):
+        output = ["Dry Run"]
+        output.append(f"File: {filepath}")
+        try:
+            input_file = CFDataset(filepath)
+        except Exception as e:
+            output.append(f"{e.__class__.__name__}: {e}")
+        else:
+            periods = input_file.climo_periods.keys() & climo
+            output.append(f"climo_periods: {periods}")
+            for attr in "project institution model emissions run".split():
+                try:
+                    output.append(f"{attr}: {getattr(input_file.metadata, attr)}")
+                except Exception as e:
+                    output.append(f"{attr}: {e.__class__.__name__}: {e}")
+            output.append(f"dependent_varnames: {input_file.dependent_varnames()}")
+            for attr in "time_resolution is_multi_year_mean".split():
+                output.append(f"{attr}: {getattr(input_file, attr)}")
+
+        filename = os.path.join(self.workdir, "dry.txt")
+        with open(filename, "w") as f:
+            for line in output:
+                f.write(f"{line}\n")
+
+        return filename
+
+    def collect_climo_files(self):
+        return [file for file in os.listdir(self.workdir) if file.endswith(".nc")]
+
+    def build_meta_link(self, climo_files):
+        meta_link = MetaLink4(
+            "outputs", "Output of netCDF climo files", workdir=self.workdir
+        )
+
+        for file in climo_files:
+            # Create a MetaFile instance, which instantiates a ComplexOutput object.
+            meta_file = MetaFile(f"{file}", "Climatology", fmt=FORMATS.NETCDF)
+            meta_file.file = os.path.join(self.workdir, file)
+            meta_link.append(meta_file)
+
+        return meta_link.xml
+
+    def format_climo(self, climo):
+        if "all" in climo:
+            return self.climos["historical"] + self.climos["futures"]
+
+        # loop over given climo, replace items ('historical', 'futures') with
+        # items from list
+        return list(
+            set(
```

Comment:
I think you can use the set comprehension operator here: `{ x for x in iterable }`.